teps/0091-trusted-resources.md

```diff
@@ -377,13 +377,18 @@ metadata:
   name: feature-flag
   namespace: tekton-pipelines
 data:
-  verification-policy: enforce
+  resource-verification-mode: verify_all
 ```
 
-`verification-policy`. (Optional, `enforce`, `warn` or `skip`, default `skip`):
- * `enforce`: Failing verification will mark the taskruns/pipelineruns as failed.
- * `warn`: Log warning but don't fail the taskruns/pipelineruns.
- * `skip`: Directly skip the verification.
+`resource-verification-mode`. (Optional, `verify_all`, `verify_match` or `no_verification`, default `no_verification`):
+ * `verify_all`: Failing verification will mark the taskruns/pipelineruns as failed.
+ * `verify_match`: Fetch VerificationPolicy from the namespace, and if policies matched then verify, otherwise skip the verification. Details explained in [ Verify the Resources](#verify-the-resources)
```

Comment:
I don't think this is a good idea to support globally, though I get why we'd want something like this for incremental adoption. It would be generally be bad if Tasks could circumvent verification just by renaming to something that's not covered by policy, and new Tasks that are probably riskier would be exempted by default since they're unlikely to match existing policies.

Instead of making verification opt-in, it would probably be safer to make the mode opt-out on a per-policy basis (see [`ClusterImagePolicySpec.mode`](https://github.com/sigstore/policy-controller/blob/main/docs/api-types/index.md#clusterimagepolicyspec) for an example). This way things are directed to the more secure way by default, forcing users to either configure the matching policy or explicitly opt out the Tasks.

Reply:
Thanks! That makes sense. By looking at the [ClusterImagePolicySpec.mode](https://github.com/sigstore/policy-controller/blob/main/docs/api-types/index.md#clusterimagepolicyspec).  How about changing the `verify_match` to:
1. If found matched polices then verify and fail
2. If no matched policies then warn

Reply:
Preferring warn over skip generally sounds good, but that doesn't address the larger problem of opt-in enforcement and the ability to evade enforcement by crafting a non-matching artifact. 😢 

Having policies that opt-out matching patterns seems like a better global policy - this way new artifacts that we haven't encountered before (whether they are malicious or not) can't bypass enforcement.